### PR TITLE
items stats: add option to show weight

### LINKED ISSUE #6939: Plugin: Items Stats (Option to remove weight)
Make an option to remove weight from item stats.

### PR DESCRIPTION
**Pull request by Dava96**
items stats: add option to show weight
resolves #6939

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/itemstats/ItemStatConfig.java [hunk 1]
```diff
@@ -79,7 +79,17 @@ default boolean absolute()
 	)
 	default boolean theoretical()
 	{
-		return false;
+		return true;
```
Comment (TheStonedTurtle) [inline runelite-client/src/main/java/net/runelite/client/plugins/itemstats/ItemStatConfig.java]: Why did you update the default state of this config option?

File: runelite-client/src/main/java/net/runelite/client/plugins/itemstats/ItemStatOverlay.java [hunk 2]
```diff
@@ -182,7 +182,9 @@ private String getChangeString(
 	private String buildStatBonusString(ItemStats s)
 	{
 		final StringBuilder b = new StringBuilder();
-		b.append(getChangeString("Weight", s.getWeight(), true, false));
+		if (config.showWeight()) {
```
Comment (TheStonedTurtle) [inline runelite-client/src/main/java/net/runelite/client/plugins/itemstats/ItemStatOverlay.java]: All brackets should be on a new line, follow our checkstyle: https://github.com/runelite/runelite/wiki/Checkstyle-with-IntelliJ-IDEA

File: runelite-client/src/main/java/net/runelite/client/plugins/itemstats/ItemStatConfig.java [hunk 3]
```diff
@@ -82,6 +82,16 @@ default boolean theoretical()
 		return false;
 	}
 
+	@ConfigItem(
+		keyName = "showWeight",
+		name = "Show Weight",
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/itemstats/ItemStatConfig.java]:
```suggestion
		name = "Hide weight",
```

File: runelite-client/src/main/java/net/runelite/client/plugins/itemstats/ItemStatConfig.java [hunk 4]
```diff
@@ -82,6 +82,16 @@ default boolean theoretical()
...
+		description = "Show weight or hide weight in the tooltip."
+	)
+	default boolean showWeight()
+	{
+		return true;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/itemstats/ItemStatConfig.java]:
```suggestion
		return false;
```

File: runelite-client/src/main/java/net/runelite/client/plugins/itemstats/ItemStatConfig.java [hunk 5]
```diff
@@ -82,6 +82,16 @@ default boolean theoretical()
 		return false;
 	}
 
+	@ConfigItem(
+		keyName = "showWeight",
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/itemstats/ItemStatConfig.java]:
```suggestion
		keyName = "hideWeight",
```

File: runelite-client/src/main/java/net/runelite/client/plugins/itemstats/ItemStatOverlay.java [hunk 6]
```diff
@@ -182,7 +182,10 @@ private String getChangeString(
 	private String buildStatBonusString(ItemStats s)
 	{
 		final StringBuilder b = new StringBuilder();
-		b.append(getChangeString("Weight", s.getWeight(), true, false));
+		if (config.showWeight())
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/itemstats/ItemStatOverlay.java]:
```suggestion
		if (!config.hideWeight())
```

**Comment by Dava96** [comment]
I believe I have addressed all your requests here?

**Comment by Adam-** [comment]
I think that the original way this was made more sense.

**Pull request merged**